src/core/Oscillator.cpp

```diff
@@ -28,9 +29,33 @@
 #include "Engine.h"
 #include "Mixer.h"
 #include "AutomatableModel.h"
+#include "fftw3.h"
 
 
 
+void Oscillator::waveTableInit()
+{
+	if (!s_waveTableBandFreqs)
+	{
+		createFFtPlans();
+		s_waveTableBandFreqs = new float[127];
+		s_waveTablesPerWaveformCount = 0;
+		for (int i = 1; i < 127; i+=4)
+		{
+			s_waveTableBandFreqs[s_waveTablesPerWaveformCount] = 440.0 * powf(2.0, (i - 69.0) / 12.0);
```

Comment:
Similar to `waveTableBandFromFreq`, I think it would be simpler to replace `s_waveTableBandFreqs` with a function that computes the frequency on-demand (actually, I'll bet there's a function somewhere in the codebase that already does this). Fewer allocations and state to keep track of.

Reply:
`s_waveTableBandFreqs` has been replaced with the `freqFromWaveTableBand(i)` function.

Researching the current codebase for similar implementations leads to https://github.com/LMMS/lmms/blob/62d505b2e6af03f3f2d63ed7f23c254fa64b9f95/src/core/NotePlayHandle.cpp#L513

The implementation in NotePlayHandle is unsuitable for use in this context, as it is designed to handle Base frequency and instrument pitching, The requirements of the oscillator is not to convert from midi, as this is already correctly done but from wave table band.